backend/open_webui/apps/retrieval/main.py

```diff
@@ -486,7 +496,11 @@ class ConfigUpdateForm(BaseModel):
 
 
 @app.post("/config/update")
-async def update_rag_config(form_data: ConfigUpdateForm, user=Depends(get_admin_user)):
+async def update_rag_config(
+    form_data: ConfigUpdateForm,
+    user=Depends(get_admin_user),
+    audit_logger: AuditLogger = Depends(get_audit_logger),
```

Comment:
Quick q: Is there a way to centrally instantiate audit_logger and import that instead of using Depends? Not using `Depends` would be preferable here as it won't require much modifications to the codebase relatively speaking. What would be the trade off for that?

Reply:
Using it as a dependency can make it easier to inject request-specific data down the line, also making it easier to test. But my main concern in the beginning was concurrency headaches (which may or may not be a problem), the idea was to keep all the dependencies in the scope of the request. Also, avoid more repetition.

Reply:
But depending on the desired result, if it is avoiding having to inject this in every call, maybe it could be set to a router level dependency, for example:
```
APIRouter(
    dependencies=[Depends(get_audit_logger)]
)
``` 
and in `get_audit_logger` store the logger instance in the state (`...state.audit_logger`)
And in each request get it from the state `audit_logger = request.state.audit_logger`

Reply:
There might be cleaner solutions, but I'm making maximum effort to avoid changing code that is already working and just adding around it